src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,42 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserLogin(user.getLogin());
+        checkUserPassword(user.getPassword());
+        checkUserAge(user.getAge());
+        return storageDao.add(user);
+    }
+
+    private void checkUserLogin(String login) {
+        if (login == null || login.isEmpty() || login.isBlank()) {
+            throw new RuntimeException("User login is null");
+        }
+        if (storageDao.get(login) != null) {
+            throw new RuntimeException("User with this login exists in storage");
+        }
+    }
+
+    private void checkUserPassword(String password) {
+        if (password == null || password.isEmpty() || password.isBlank()) {
+            throw new RuntimeException("User password is null");
```

Comment:
Statement that it's null is not fully correct, better to have more generic message in this case
```suggestion
            throw new RuntimeException("User password wasn't provided");
```